// angular2-baidu-map/esm5/components/canvaslayer.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Directive, EventEmitter, Input, Output } from '@angular/core';
import { MapService } from '../providers/mapService';
import { isNumber } from '../helpers/object';
var CanvasLayerComponent = /** @class */ (function () {
    function CanvasLayerComponent(_service) {
        this._service = _service;
        this.loaded = new EventEmitter();
    }
    /**
     * @return {?}
     */
    CanvasLayerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this._service
            .addOverlay(function (map) {
            _this.canvaslayer = new window.BMap.CanvasLayer(_this.getOptions(_this.options, map));
            return _this.canvaslayer;
        })
            .then(function () {
            _this.loaded.emit(_this.canvaslayer);
        });
    };
    /**
     * @return {?}
     */
    CanvasLayerComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this._service.removeOverlay(this.canvaslayer);
    };
    /**
     * @param {?} options
     * @param {?} map
     * @return {?}
     */
    CanvasLayerComponent.prototype.getOptions = /**
     * @param {?} options
     * @param {?} map
     * @return {?}
     */
    function (options, map) {
        if (!options) {
            return;
        }
        var /** @type {?} */ opts = {};
        if (isNumber(options.zIndex)) {
            opts.zIndex = options.zIndex;
        }
        if (options.paneName) {
            opts.paneName = options.paneName;
        }
        if (options.update) {
            opts.update = function () {
                return options.update(map, this.canvas);
            };
        }
        return opts;
    };
    CanvasLayerComponent.decorators = [
        { type: Directive, args: [{
                    selector: 'canvaslayer'
                },] },
    ];
    /** @nocollapse */
    CanvasLayerComponent.ctorParameters = function () { return [
        { type: MapService, },
    ]; };
    CanvasLayerComponent.propDecorators = {
        "options": [{ type: Input },],
        "loaded": [{ type: Output },],
    };
    return CanvasLayerComponent;
}());
export { CanvasLayerComponent };
function CanvasLayerComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    CanvasLayerComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    CanvasLayerComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    CanvasLayerComponent.propDecorators;
    /** @type {?} */
    CanvasLayerComponent.prototype.options;
    /** @type {?} */
    CanvasLayerComponent.prototype.loaded;
    /** @type {?} */
    CanvasLayerComponent.prototype.canvaslayer;
    /** @type {?} */
    CanvasLayerComponent.prototype._service;
}
