// angular2-baidu-map/esm2015/components/canvaslayer.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Directive, EventEmitter, Input, Output } from '@angular/core';
import { MapService } from '../providers/mapService';
import { isNumber } from '../helpers/object';
export class CanvasLayerComponent {
    /**
     * @param {?} _service
     */
    constructor(_service) {
        this._service = _service;
        this.loaded = new EventEmitter();
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this._service
            .addOverlay((map) => {
            this.canvaslayer = new window.BMap.CanvasLayer(this.getOptions(this.options, map));
            return this.canvaslayer;
        })
            .then(() => {
            this.loaded.emit(this.canvaslayer);
        });
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        this._service.removeOverlay(this.canvaslayer);
    }
    /**
     * @param {?} options
     * @param {?} map
     * @return {?}
     */
    getOptions(options, map) {
        if (!options) {
            return;
        }
        const /** @type {?} */ opts = {};
        if (isNumber(options.zIndex)) {
            opts.zIndex = options.zIndex;
        }
        if (options.paneName) {
            opts.paneName = options.paneName;
        }
        if (options.update) {
            opts.update = function () {
                return options.update(map, this.canvas);
            };
        }
        return opts;
    }
}
CanvasLayerComponent.decorators = [
    { type: Directive, args: [{
                selector: 'canvaslayer'
            },] },
];
/** @nocollapse */
CanvasLayerComponent.ctorParameters = () => [
    { type: MapService, },
];
CanvasLayerComponent.propDecorators = {
    "options": [{ type: Input },],
    "loaded": [{ type: Output },],
};
function CanvasLayerComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    CanvasLayerComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    CanvasLayerComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    CanvasLayerComponent.propDecorators;
    /** @type {?} */
    CanvasLayerComponent.prototype.options;
    /** @type {?} */
    CanvasLayerComponent.prototype.loaded;
    /** @type {?} */
    CanvasLayerComponent.prototype.canvaslayer;
    /** @type {?} */
    CanvasLayerComponent.prototype._service;
}
