// angular2-baidu-map/esm5/types/Map.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { isFunction } from '../helpers/object';
/**
 * @record
 */
export function BMapConstructor() { }
function BMapConstructor_tsickle_Closure_declarations() {
    /* TODO: handle strange member:
    new (el: HTMLElement | string, opts: MapOptions): BMapInstance
    */
}
/**
 * @record
 */
export function BMapInstance() { }
function BMapInstance_tsickle_Closure_declarations() {
    /** @type {?} */
    BMapInstance.prototype.disableDragging;
    /** @type {?} */
    BMapInstance.prototype.enableDragging;
    /** @type {?} */
    BMapInstance.prototype.disableScrollWheelZoom;
    /** @type {?} */
    BMapInstance.prototype.enableScrollWheelZoom;
    /** @type {?} */
    BMapInstance.prototype.disableDoubleClickZoom;
    /** @type {?} */
    BMapInstance.prototype.enableDoubleClickZoom;
    /** @type {?} */
    BMapInstance.prototype.disableKeyboard;
    /** @type {?} */
    BMapInstance.prototype.enableKeyboard;
    /** @type {?} */
    BMapInstance.prototype.disableInertialDragging;
    /** @type {?} */
    BMapInstance.prototype.enableInertialDragging;
    /** @type {?} */
    BMapInstance.prototype.disableContinuousZoom;
    /** @type {?} */
    BMapInstance.prototype.enableContinuousZoom;
    /** @type {?} */
    BMapInstance.prototype.disablePinchToZoom;
    /** @type {?} */
    BMapInstance.prototype.enablePinchToZoom;
    /** @type {?} */
    BMapInstance.prototype.disableAutoResize;
    /** @type {?} */
    BMapInstance.prototype.enableAutoResize;
    /** @type {?} */
    BMapInstance.prototype.addControl;
    /** @type {?} */
    BMapInstance.prototype.removeControl;
    /** @type {?} */
    BMapInstance.prototype.addOverlay;
    /** @type {?} */
    BMapInstance.prototype.removeOverlay;
    /** @type {?} */
    BMapInstance.prototype.clearOverlays;
    /** @type {?} */
    BMapInstance.prototype.addTileLayer;
    /** @type {?} */
    BMapInstance.prototype.removeTileLayer;
    /** @type {?} */
    BMapInstance.prototype.setDefaultCursor;
    /** @type {?} */
    BMapInstance.prototype.getDefaultCursor;
    /** @type {?} */
    BMapInstance.prototype.setDraggingCursor;
    /** @type {?} */
    BMapInstance.prototype.getDraggingCursor;
    /** @type {?} */
    BMapInstance.prototype.setMinZoom;
    /** @type {?} */
    BMapInstance.prototype.setMaxZoom;
    /** @type {?} */
    BMapInstance.prototype.getBounds;
    /** @type {?} */
    BMapInstance.prototype.setCenter;
    /** @type {?} */
    BMapInstance.prototype.getCenter;
    /** @type {?} */
    BMapInstance.prototype.getDistance;
    /** @type {?} */
    BMapInstance.prototype.setCurrentCity;
    /** @type {?} */
    BMapInstance.prototype.centerAndZoom;
    /** @type {?} */
    BMapInstance.prototype.setMapType;
    /** @type {?} */
    BMapInstance.prototype.getMapType;
    /** @type {?} */
    BMapInstance.prototype.getSize;
    /** @type {?} */
    BMapInstance.prototype.setZoom;
    /** @type {?} */
    BMapInstance.prototype.getZoom;
    /** @type {?} */
    BMapInstance.prototype.panTo;
    /** @type {?} */
    BMapInstance.prototype.panBy;
    /** @type {?} */
    BMapInstance.prototype.reset;
    /** @type {?} */
    BMapInstance.prototype.highResolutionEnabled;
    /** @type {?} */
    BMapInstance.prototype.zoomIn;
    /** @type {?} */
    BMapInstance.prototype.zoomOut;
    /** @type {?} */
    BMapInstance.prototype.getContainer;
    /** @type {?} */
    BMapInstance.prototype.closeInfoWindow;
    /** @type {?} */
    BMapInstance.prototype.pixelToPoint;
    /** @type {?} */
    BMapInstance.prototype.pointToPixel;
    /** @type {?} */
    BMapInstance.prototype.openInfoWindow;
    /** @type {?} */
    BMapInstance.prototype.addEventListener;
    /** @type {?} */
    BMapInstance.prototype.removeEventListener;
}
/**
 * @record
 */
export function MapOptions() { }
function MapOptions_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    MapOptions.prototype.minZoom;
    /** @type {?|undefined} */
    MapOptions.prototype.maxZoom;
    /** @type {?|undefined} */
    MapOptions.prototype.enableHighResolution;
    /** @type {?|undefined} */
    MapOptions.prototype.enableAutoResize;
    /** @type {?|undefined} */
    MapOptions.prototype.enableMapClick;
    /** @type {?|undefined} */
    MapOptions.prototype.disableDragging;
    /** @type {?|undefined} */
    MapOptions.prototype.enableScrollWheelZoom;
    /** @type {?|undefined} */
    MapOptions.prototype.disableDoubleClickZoom;
    /** @type {?|undefined} */
    MapOptions.prototype.enableKeyboard;
    /** @type {?|undefined} */
    MapOptions.prototype.enableInertialDragging;
    /** @type {?|undefined} */
    MapOptions.prototype.enableContinuousZoom;
    /** @type {?|undefined} */
    MapOptions.prototype.disablePinchToZoom;
    /** @type {?|undefined} */
    MapOptions.prototype.cursor;
    /** @type {?|undefined} */
    MapOptions.prototype.draggingCursor;
    /** @type {?|undefined} */
    MapOptions.prototype.currentCity;
    /** @type {?|undefined} */
    MapOptions.prototype.centerAndZoom;
    /** @type {?|undefined} */
    MapOptions.prototype.mapType;
}
/**
 * @record
 */
export function CenterAndZoom() { }
function CenterAndZoom_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    CenterAndZoom.prototype.zoom;
}
/**
 * @record
 */
export function BPanOptions() { }
function BPanOptions_tsickle_Closure_declarations() {
    /** @type {?} */
    BPanOptions.prototype.noAnimation;
}
/** @enum {string} */
var MapTypeEnum = {
    BMAP_NORMAL_MAP: 'BMAP_NORMAL_MAP',
    BMAP_PERSPECTIVE_MAP: 'BMAP_PERSPECTIVE_MAP',
    BMAP_SATELLITE_MAP: 'BMAP_SATELLITE_MAP',
    BMAP_HYBRID_MAP: 'BMAP_HYBRID_MAP',
};
export { MapTypeEnum };
/**
 * @param {?} maptype
 * @return {?}
 */
export function isMapTypeEnum(maptype) {
    return !isFunction((/** @type {?} */ (maptype)).getTileLayer);
}
