// angular2-baidu-map/esm2015/types/CanvasLayer.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * @record
 */
export function BCanvasLayerConstructor() { }
function BCanvasLayerConstructor_tsickle_Closure_declarations() {
    /* TODO: handle strange member:
    new (options?: BCanvasLayerOptions): BCanvasLayer
    */
}
/**
 * @record
 */
export function CanvasLayerOptions() { }
function CanvasLayerOptions_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    CanvasLayerOptions.prototype.zIndex;
    /** @type {?|undefined} */
    CanvasLayerOptions.prototype.paneName;
    /** @type {?|undefined} */
    CanvasLayerOptions.prototype.update;
}
/**
 * @record
 */
export function BCanvasLayerOptions() { }
function BCanvasLayerOptions_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    BCanvasLayerOptions.prototype.zIndex;
    /** @type {?|undefined} */
    BCanvasLayerOptions.prototype.paneName;
    /** @type {?|undefined} */
    BCanvasLayerOptions.prototype.update;
}
/**
 * @record
 */
export function BCanvasLayer() { }
function BCanvasLayer_tsickle_Closure_declarations() {
}
