// angular2-baidu-map/esm2015/types/InfoWindow.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * @record
 */
export function BInfoWindowConstructor() { }
function BInfoWindowConstructor_tsickle_Closure_declarations() {
    /* TODO: handle strange member:
    new (content: string | HTMLElement, opts?: BInfoWindowOptions): BInfoWindow
    */
}
/**
 * @record
 */
export function BInfoWindow() { }
function BInfoWindow_tsickle_Closure_declarations() {
    /** @type {?} */
    BInfoWindow.prototype.setWidth;
    /** @type {?} */
    BInfoWindow.prototype.setHeight;
    /** @type {?} */
    BInfoWindow.prototype.redraw;
    /** @type {?} */
    BInfoWindow.prototype.setTitle;
    /** @type {?} */
    BInfoWindow.prototype.getTitle;
    /** @type {?} */
    BInfoWindow.prototype.setContent;
    /** @type {?} */
    BInfoWindow.prototype.getContent;
    /** @type {?} */
    BInfoWindow.prototype.getPosition;
    /** @type {?} */
    BInfoWindow.prototype.enableMaximize;
    /** @type {?} */
    BInfoWindow.prototype.disableMaximize;
    /** @type {?} */
    BInfoWindow.prototype.isOpen;
    /** @type {?} */
    BInfoWindow.prototype.setMaxContent;
    /** @type {?} */
    BInfoWindow.prototype.maximize;
    /** @type {?} */
    BInfoWindow.prototype.restore;
    /** @type {?} */
    BInfoWindow.prototype.enableAutoPan;
    /** @type {?} */
    BInfoWindow.prototype.disableAutoPan;
    /** @type {?} */
    BInfoWindow.prototype.enableCloseOnClick;
    /** @type {?} */
    BInfoWindow.prototype.disableCloseOnClick;
    /** @type {?} */
    BInfoWindow.prototype.addEventListener;
    /** @type {?} */
    BInfoWindow.prototype.removeEventListener;
}
/**
 * @record
 */
export function BInfoWindowOptions() { }
function BInfoWindowOptions_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    BInfoWindowOptions.prototype.width;
    /** @type {?|undefined} */
    BInfoWindowOptions.prototype.height;
    /** @type {?|undefined} */
    BInfoWindowOptions.prototype.maxWidth;
    /** @type {?|undefined} */
    BInfoWindowOptions.prototype.offset;
    /** @type {?|undefined} */
    BInfoWindowOptions.prototype.title;
    /** @type {?|undefined} */
    BInfoWindowOptions.prototype.enableAutoPan;
    /** @type {?|undefined} */
    BInfoWindowOptions.prototype.enableCloseOnClick;
    /** @type {?|undefined} */
    BInfoWindowOptions.prototype.enableMessage;
    /** @type {?|undefined} */
    BInfoWindowOptions.prototype.message;
}
