// angular2-baidu-map/esm2015/types/Marker.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * @record
 */
export function BMarkerConstructor() { }
function BMarkerConstructor_tsickle_Closure_declarations() {
    /* TODO: handle strange member:
    new (point: BPoint, options?: BMarkerOptions): BMarker
    */
}
/**
 * @record
 */
export function BMarker() { }
function BMarker_tsickle_Closure_declarations() {
    /** @type {?} */
    BMarker.prototype.addEventListener;
    /** @type {?} */
    BMarker.prototype.removeEventListener;
    /** @type {?} */
    BMarker.prototype.setPosition;
    /** @type {?} */
    BMarker.prototype.setOffset;
    /** @type {?} */
    BMarker.prototype.setIcon;
    /** @type {?} */
    BMarker.prototype.getIcon;
    /** @type {?} */
    BMarker.prototype.enableMassClear;
    /** @type {?} */
    BMarker.prototype.disableMassClear;
    /** @type {?} */
    BMarker.prototype.enableDragging;
    /** @type {?} */
    BMarker.prototype.disableDragging;
    /** @type {?} */
    BMarker.prototype.setRotation;
    /** @type {?} */
    BMarker.prototype.setShadow;
    /** @type {?} */
    BMarker.prototype.setTitle;
    /** @type {?} */
    BMarker.prototype.setLabel;
    /** @type {?} */
    BMarker.prototype.getLabel;
    /** @type {?} */
    BMarker.prototype.setTop;
    /** @type {?} */
    BMarker.prototype.setAnimation;
    /** @type {?} */
    BMarker.prototype.getPosition;
}
/**
 * @record
 */
export function Marker() { }
function Marker_tsickle_Closure_declarations() {
    /** @type {?} */
    Marker.prototype.point;
    /** @type {?|undefined} */
    Marker.prototype.options;
}
/**
 * @record
 */
export function MarkerOptions() { }
function MarkerOptions_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    MarkerOptions.prototype.offset;
    /** @type {?|undefined} */
    MarkerOptions.prototype.icon;
    /** @type {?|undefined} */
    MarkerOptions.prototype.enableMassClear;
    /** @type {?|undefined} */
    MarkerOptions.prototype.enableDragging;
    /** @type {?|undefined} */
    MarkerOptions.prototype.enableClicking;
    /** @type {?|undefined} */
    MarkerOptions.prototype.raiseOnDrag;
    /** @type {?|undefined} */
    MarkerOptions.prototype.draggingCursor;
    /** @type {?|undefined} */
    MarkerOptions.prototype.rotation;
    /** @type {?|undefined} */
    MarkerOptions.prototype.shadow;
    /** @type {?|undefined} */
    MarkerOptions.prototype.title;
}
/**
 * @record
 */
export function BMarkerOptions() { }
function BMarkerOptions_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    BMarkerOptions.prototype.offset;
    /** @type {?|undefined} */
    BMarkerOptions.prototype.icon;
    /** @type {?|undefined} */
    BMarkerOptions.prototype.enableMassClear;
    /** @type {?|undefined} */
    BMarkerOptions.prototype.enableDragging;
    /** @type {?|undefined} */
    BMarkerOptions.prototype.enableClicking;
    /** @type {?|undefined} */
    BMarkerOptions.prototype.raiseOnDrag;
    /** @type {?|undefined} */
    BMarkerOptions.prototype.draggingCursor;
    /** @type {?|undefined} */
    BMarkerOptions.prototype.rotation;
    /** @type {?|undefined} */
    BMarkerOptions.prototype.shadow;
    /** @type {?|undefined} */
    BMarkerOptions.prototype.title;
}
