// angular2-baidu-map/esm2015/helpers/object.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * @param {?} obj
 * @return {?}
 */
export function isNull(obj) {
    return obj === null || obj === undefined;
}
/**
 * @param {?} obj
 * @return {?}
 */
export function isUndefined(obj) {
    return obj === undefined;
}
/**
 * @param {?} obj
 * @return {?}
 */
export function isBoolean(obj) {
    return Object.prototype.toString.call(obj) === '[object Boolean]';
}
/**
 * @param {?} obj
 * @return {?}
 */
export function isFunction(obj) {
    return Object.prototype.toString.call(obj) === '[object Function]';
}
/**
 * @param {?} obj
 * @return {?}
 */
export function isString(obj) {
    return Object.prototype.toString.call(obj) === '[object String]';
}
/**
 * @param {?} obj
 * @return {?}
 */
export function isNumber(obj) {
    return Object.prototype.toString.call(obj) === '[object Number]';
}
/**
 * @template T
 * @param {?} obj
 * @param {...?} keys
 * @return {?}
 */
export function omit(obj, ...keys) {
    const /** @type {?} */ rawKeys = Object.keys(obj);
    const /** @type {?} */ finalKeys = rawKeys.filter(k => keys.indexOf(k) < 0);
    return finalKeys.reduce((p, v) => {
        p[v] = obj[v];
        return p;
    }, /** @type {?} */ ({}));
}
