// angular2-baidu-map/esm5/index.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { NgModule } from '@angular/core';
import { ControlComponent } from './components/control.component';
import { MapComponent } from './components/map.component';
import { MarkerComponent } from './components/marker.component';
import { PolylineComponent } from './components/polyline.component';
import { CircleComponent } from './components/circle.component';
import { PolygonComponent } from './components/polygon.component';
import { HeatmapComponent } from './components/heatmap.component';
import { TileLayerComponent } from './components/tilelayer.component';
import { TrafficLayerComponent } from './components/trafficlayer.component';
import { CanvasLayerComponent } from './components/canvaslayer.component';
import { MarkerClustererComponent } from './components/markerClusterer.component';
import { ScriptLoader, ScriptLoaderConfig } from './providers/scriptLoader';
var BaiduMapModule = /** @class */ (function () {
    function BaiduMapModule() {
    }
    /**
     * @param {?=} _config
     * @return {?}
     */
    BaiduMapModule.forRoot = /**
     * @param {?=} _config
     * @return {?}
     */
    function (_config) {
        return {
            ngModule: BaiduMapModule,
            providers: [{ provide: ScriptLoaderConfig, useValue: _config }, ScriptLoader]
        };
    };
    BaiduMapModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [
                        MapComponent,
                        MarkerComponent,
                        ControlComponent,
                        PolylineComponent,
                        CircleComponent,
                        PolygonComponent,
                        HeatmapComponent,
                        TileLayerComponent,
                        TrafficLayerComponent,
                        CanvasLayerComponent,
                        MarkerClustererComponent
                    ],
                    exports: [
                        MapComponent,
                        MarkerComponent,
                        ControlComponent,
                        PolylineComponent,
                        CircleComponent,
                        PolygonComponent,
                        HeatmapComponent,
                        TileLayerComponent,
                        TrafficLayerComponent,
                        CanvasLayerComponent,
                        MarkerClustererComponent
                    ]
                },] },
    ];
    return BaiduMapModule;
}());
export { BaiduMapModule };
function BaiduMapModule_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    BaiduMapModule.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    BaiduMapModule.ctorParameters;
}
export { MapTypeEnum } from './types/Map';
export { ControlAnchor, LengthUnit, NavigationControlType, MapTypeControlType } from './types/Control';
export { Animation } from './types/Animation';
