// angular2-baidu-map/esm2015/angular2-baidu-map.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Generated bundle index. Do not edit.
 */
export { BaiduMapModule, MapTypeEnum, ControlAnchor, LengthUnit, NavigationControlType, MapTypeControlType, Animation } from './public_api';
export { CanvasLayerComponent as ɵm } from './components/canvaslayer.component';
export { CircleComponent as ɵh } from './components/circle.component';
export { ControlComponent as ɵf } from './components/control.component';
export { HeatmapComponent as ɵj } from './components/heatmap.component';
export { MapComponent as ɵa } from './components/map.component';
export { MarkerComponent as ɵe } from './components/marker.component';
export { MarkerClustererComponent as ɵn } from './components/markerClusterer.component';
export { PolygonComponent as ɵi } from './components/polygon.component';
export { PolylineComponent as ɵg } from './components/polyline.component';
export { TileLayerComponent as ɵk } from './components/tilelayer.component';
export { TrafficLayerComponent as ɵl } from './components/trafficlayer.component';
export { MapService as ɵb } from './providers/mapService';
export { ScriptLoader as ɵd, ScriptLoaderConfig as ɵc } from './providers/scriptLoader';
